include/swift/AST/Decl.h

```diff
@@ -667,6 +667,15 @@ class alignas(1 << DeclAlignInBits) Decl {
   void operator=(const Decl&) = delete;
   SourceLoc getLocFromSource() const;
 
+  struct CachedExternalSourceLocs {
+    SourceLoc Loc;
+    SourceLoc StartLoc;
+    SourceLoc EndLoc;
+    // We need to keep this user-defined constructor build on linux.
+    CachedExternalSourceLocs() {}
```

Comment:
Wow, that definitely shouldn't be necessary. (I believe you though.) Do you still remember what the error message was?

Reply:
It seems the explicit constructor isn't necessary. All we need is to add `{}` after the `CachedExternalSourceLocs` data, see here: https://stackoverflow.com/questions/7411515/why-does-c-require-a-user-provided-default-constructor-to-default-construct-a

Reply:
Ah, I guess that makes sense. Zero-initializing wouldn't be correct, so you actually have to call the constructor explicitly. Or something.